src/wasm/wat-parser.cpp

```diff
@@ -1161,7 +1162,7 @@ struct ParseModuleTypesCtx : TypeParserCtx<ParseModuleTypesCtx>,
     : TypeParserCtx<ParseModuleTypesCtx>(typeIndices), in(in), wasm(wasm),
       types(types), implicitTypes(implicitTypes) {}
 
-  Result<HeapTypeT> getHeapTypeFromIdx(Index idx) {
+  Result<HeapTypeT> getHeapTypeIdxFromInt(Index idx) {
```

Comment:
This returns a `HeapType`, doesn't it, not an index? At least in some code paths we have `HeapTypeT = HeapType`.

Reply:
Yes, but in Binaryen IR the "index" type used to refer to a heap type is `HeapType`. Basically the concrete types here are completely irrelevant; we're talking about indices in the sense of the Wasm abstract semantics here.

Reply:
Hmm, I see. I guess in that case I'm really not sure if this is a significant improvement. The reader needs to keep a bunch of stuff in their head both before this PR and after, that is. So I'd be ok without this PR - what do you think?

Reply:
Yeah, I don't feel strongly about it. Will close for now.